algebra-core/mince/src/intrinsics.rs

```diff
@@ -0,0 +1,46 @@
+use proc_macro::TokenStream;
+use quote::quote;
+
+pub fn define_intrinsics() -> TokenStream {
+    (quote! {
+        {
+            let mut begin = || {
+                asm_string.replace_with(|_| "\"".to_string());
+            };
```

Comment:
Hmm instead of doing this, why not just append to the string in place? I.e. something like `asm_string.borrow_mut().push_str(...)`.

Reply:
(Similarly for the other intrinsics)

Reply:
I believe this was because closures had an extremely limiting feature called unique borrowing, disallowing multiple closures from borrowing the same underlying mutable object, that forced me to use a refCell. This was one of the times I was really frustrated with the borrow checker.

Reply:
Right, I understand why you went with `RefCell`. I was saying that we should take advantage of `RefCell` more.

In more detail, `RefCell` essentially enforces the borrow-checker rules only at run-time: it says that you can have multiple mutable references to an item, but you must only use one of those mutable references at a time. 

This is exactly how we're using `asm_string`: once we mutate it, we immediately drop the mutation "lock" that we acquired, and so we're upholding the invariant of `RefCell`.

Reply:
Push_str seems to work only for str whereas I require format! which produces a String. This seemed like the most straightforward and ergonomic way to mutate the string. However, I did find I could get by without the clone() s. 

If there are no strong reasons to change it, I would prefer to keep it the way it is rather than to fiddle further.

Reply:
You can do `&format!(...)` to produce a string. The primary reason to change it would be clarify what's going on; it took me a couple of minutes to understand what was going on.

Reply:
That being said, we can just file an issue for this and then revisit in a later PR.